### ISSUE TITLE
Support fetching changelog from bitbucket-server

### ISSUE THREAD

**Issue by zeldigas** [renovatebot/renovate#8662]
**What would you like Renovate to be able to do?**

Right now repositories stored in BitbucketServer are not processed during release-notes fetching, I got the following debug message in logs:
```
DEBUG: Fetching changelog: https://mybb.acme.com/projects/PRJ/repos/internal-repo/browse (3.2.0 -> 3.3.0) (repository=~jsmith/changelog-test)
DEBUG: Repository URL does not match any known github hosts - skipping changelog retrieval (repository=~jsmith/changelog-test)
       "manager": "maven",
       "depName": "com.acme.libs:my-lib",
       "sourceUrl": "https://mybb.acme.com/projects/PRJ/repos/internal-repo/browse"
```

In case of configured platform to bitbucket-server I'd like renovate to support going to repository by resolved url and fetch release notes from CHANGELOG.md if it's located in repo.

**Did you already have any implementation ideas?**

Looks like current code responsible for fetching changelog is full of `if` checks for platform - for github and gitlab. I think that adding one more if branch in all the places would increase entropy significantly. I think that better option would be to introduce some interface that can be implemented by various platforms.

I can try to work on this, but might need some help from renovate dev team in case of questions as code base for PR resolution is not that small + relies on concept of releases that some platforms do not have (bitbucket-server specifically) that would affect the way to identify the range of looked up items in changelog.md file.

**Comment by viceice** [comment]
I think we have an open issue for supporting other platform changelog fetching. That code needs a major refactoring before we can add new platforms for sure.

**Comment by HonkingGoose** [comment]
I think I found the issue that @viceice is talking about, and I think the current issue is a duplicate of the older issue.

Duplicate of #3903

**Comment by viceice** [comment]
Duplicate of #3903
